components/Common/AvatarGroup/Avatar/index.tsx

```diff
@@ -0,0 +1,29 @@
+import type { FC } from 'react';
+import * as RadixAvatar from '@radix-ui/react-avatar';
+import styles from './index.module.css';
+
+type AvatarProps = {
+  src?: string;
+  alt?: string;
+  extraClass?: string;
+  onClick?: () => void;
+};
+
+const usrnameToFallbackText = (name?: string) => {
+  // sanitize name before using it
+  if (!name) return;
+  const sanitizeName = name.toLowerCase().replace(' ', '-');
+  const words = sanitizeName.split('-');
+  return words.length > 1 ? words[0][0] + words[1][0] : name.slice(0, 2);
+};
+
+const Avatar: FC<AvatarProps> = ({ src, alt, extraClass, onClick }) => (
+  <RadixAvatar.Root className={extraClass} onClick={onClick}>
+    <RadixAvatar.Image src={src} alt={alt} className={styles.avatar} />
+    <RadixAvatar.Fallback className={styles.avatar}>
+      {src ? usrnameToFallbackText(alt) : alt}
```

Comment:
What is this condition doing? Shouldn't it always invoke the function?

Reply:
this allowing to use `Avatar` for expending. So alt is `+X`

Reply:
Wait, you're using the same Avatar component for `+X`? I... gotcha. But implying that this should be done based on the presence or absence of a `src` is flakey logic.

No one besides the maintainers of this code will know that this implies that. Imagine I add a "src" on-demand? (lazy loading) then, my whole component would be just the "alt" text and I would not know why.

I discourage doing this within this component.

You should do the `+X` logic within `AvatarGroups` because that's where this makes sense. Avatar itself should not be aware of such behaviour.

Recommendation:

- Remove this `src` check. Always call the method
- Within `AAvatarGroup` you import the styles from the individual Avatar (as `import avatarStyles from './Avatar/index.css'`;
- Add the "+X" element with the Avatar styles there. That should be good enough IMO.

Or you simply remove the `usernameToFallbackText` (this is a poor method name) from the Avatar component. Which makes more sense, simply render the `alt`.

And let end-users manually call the function if they want their alt to be rendered that way. (ie: export this method to a string utility file) and simply call it outside of AvatarGroup `([username: 'ovflowd', alt: textToAcronym('ovflowd', 2)]`

And the method could become generic such as `textToAcronym` with two args: the input, and the length of the acronym (i.e. two digits)

Reply:
i have done something look clean for me. Tell me what do you think